src/tasks/set-connection-variables/set-connection-variables-v0/task.json

```diff
@@ -0,0 +1,94 @@
+{
+  "id": "d91997db-d6a0-42ed-ad3a-954b80e043ef",
+  "name": "PowerPlatformSetConnectionVariables",
+  "friendlyName": "Power Platform Set Connection Variables",
+  "description": "Sets BuildTools.* variables to provide custom script tasks access to use the Service Connection as a single source of truth.",
+  "author": "Microsoft",
+  "helpMarkDown": "[More Info](https://aka.ms/buildtoolsdoc)",
+  "category": "Utility",
+  "visibility": [
+    "Build",
+    "Release"
+  ],
+  "demands": [
+    "PowerPlatformTool"
+  ],
+  "version": {
+    "Major": "0",
+    "Minor": "1",
+    "Patch": "0"
+  },
+  "preview": false,
+  "minimumAgentVersion": "1.95.0",
+  "instanceNameFormat": "Power Platform Set Connection Variables $(message)",
+  "groups": [
+    {
+      "name": "advanced",
+      "displayName": "Advanced",
+      "isExpanded": false
+    }
+  ],
+  "inputs": [
+    {
+      "name": "authenticationType",
+      "type": "radio",
+      "label": "Authentication type",
+      "defaultValue": "PowerPlatformEnvironment",
+      "required": true,
+      "helpMarkDown": "Authentication via username/password is simpler to set up but won't work if your AAD tenant requires MFA; use Service Principal Name (SPN) if MFA is required.",
+      "options": {
+        "PowerPlatformEnvironment": "Username/password (no MFA support)",
+        "PowerPlatformSPN": "Service Principal/client secret (supports MFA)"
+      }
+    },
+    {
+      "name": "PowerPlatformEnvironment",
+      "type": "connectedService:Generic",
+      "label": "Service connection",
+      "required": true,
+      "visibleRule": "authenticationType = PowerPlatformEnvironment",
+      "helpMarkDown": "Authenticate with your Power Platform environment with username/password. Does not support MFA."
+    },
+    {
+      "name": "PowerPlatformSPN",
+      "type": "connectedService:powerplatform-spn",
+      "label": "Service connection",
+      "required": true,
+      "visibleRule": "authenticationType = PowerPlatformSPN",
+      "helpMarkDown": "Authenticate with your Power Platform environment with an Azure AppID, tenantID and client secret."
+    },
+    {
+      "name": "Environment",
+      "label": "Environment Url",
+      "type": "string",
+      "defaultValue": "$(BuildTools.EnvironmentUrl)",
+      "required": false,
+      "helpMarkDown": "Environment url this task targets; default is the pipeline variable that e.g. the CreateEnvironment task has set before."
+    },
+    {
+      "name": "ApplicationId",
+      "type": "string",
+      "label": "Application Id",
+      "defaultValue": "51f81489-12ee-4a9e-aaae-a2591f45987d",
```

Comment:
default value must not be a specific appID -> best to not have a default value

Reply:
I do recognize that is the appID we have in our documentation as a sample appID. But for this AzDevOps pipeline audience, they do need to acquire their own appID

Reply:
Ok, the idea was that using the appID we have in or docs would enable "just works" without requiring your own appID ahead.  Will fix.

Reply:
I'll also remove the advanced group so the inputs are front and center.